internal/smtpd/smtpd.go

```diff
@@ -298,23 +326,25 @@ func (srv *Server) configureDefaults() {
 	if srv.WelcomeMessage == "" {
 		srv.WelcomeMessage = fmt.Sprintf("%s ESMTP ready.", srv.Hostname)
 	}
-
 }
 
-func (session *session) serve() {
-
+func (session *session) serve(ctx context.Context) {
 	defer session.close()
 
+	if ctx.Err() != nil {
```

Comment:
What's the point of this? Were stopping early and not continuing with the current session if the context was cancelled?

Reply:
yeah - if the context is cancelled before the main serving starts, it's best to just give up right away. It would fail to deliver later anyway.